### PR TITLE
Bugfix for #168: basedir restriction and handling speed=-1 result

### PR DESCRIPTION
**Pull request by sausix**
Fix for #168 with scandir instead of glob.

Minor fix for better handling `/sys/class/net/*/speed` returning `-1`.

### CODE REVIEW

File: lib/OperatingSystems/DefaultOs.php [hunk 1]
```diff
@@ -143,19 +143,19 @@ public function getNetworkInfo() {
...
-			$iface['interface'] = basename($interface);
+			$iface['interface'] = $interface;
 			$iface['mac']       = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "link/ether " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
 			$iface['ipv4']      = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "inet " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
 			$iface['ipv6']      = shell_exec('ip -o -6 addr show ' . $iface['interface'] . ' | sed -e \'s/^.*inet6 \([^ ]\+\).*/\1/\'');
 			if ($iface['interface'] !== 'lo') {
 				$iface['status'] = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/operstate');
 				$iface['speed']  = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/speed');
-				if ($iface['speed'] !== '') {
+				if ($iface['speed'] !== '' and substr($iface['speed'], 0, 1) != "-") {
```
Comment (kesselb) [inline lib/OperatingSystems/DefaultOs.php:158]:
```suggestion
				if ($iface['speed'] !== '' && $iface['speed'] !== '-1') {
```

File: lib/OperatingSystems/DefaultOs.php [hunk 2]
```diff
@@ -143,19 +143,19 @@ public function getNetworkInfo() {
 	 * @return string
 	 */
 	public function getNetworkInterfaces() {
-		$interfaces = glob('/sys/class/net/*');
+		$interfaces = array_diff(scandir("/sys/class/net"), ['.', '..']);  // remove dot directories
```
Comment (kesselb) [inline lib/OperatingSystems/DefaultOs.php:146]:
```suggestion
		$interfaces = array_diff(scandir('/sys/class/net'), ['.', '..']);  // remove dot directories
```

**Review by kesselb — APPROVED** [review]

**Review by kesselb — CHANGES_REQUESTED** [review]
Please adjust the code like suggested and sign off the commits. Thanks!

File: lib/OperatingSystems/DefaultOs.php (hunk 1 above)
Comment (sausix) [reply lib/OperatingSystems/DefaultOs.php]: Your suggestion did not work. Have tested it with ` !== '-1'`.
Content of /sys/class/net/*/speed results in "`-1\x0a`".
```
root@nextcloud2:/sys/class/net$ hexdump -C ens3/speed
00000000  2d 31 0a                                          |-1.|
```
I hope an extra trim is ok as solution.

I've spent hours on git now. I'm not yet git aware. Sorry for the mess and the force push. Got DCO fixed at least.

Comment (kesselb) [reply lib/OperatingSystems/DefaultOs.php]: Thanks

**Comment by kesselb** [comment]
cc @tuxmaster5000 @HigH-HawK mind to test?

**Comment by HigH-HawK** [comment]
@kesselb 

Release 19.0.6:

Changes look good to me 👍

**Comment by kesselb** [comment]
Thanks for your pull request :+1: 

I assume the logic to filter out `.` and `..` is covered by https://github.com/nextcloud/serverinfo/pull/411.

If you still see the error with speed -1 please submit a new pull request.

**Pull request closed without merging**